File: frontend/lib/models/board.dart
import 'defect.dart';
import 'rework.dart';

class Board {
  final String boardId;
  final String batch;
  final DateTime inspectionTime;
  final String status; // "Passed", "Failed"
  final DateTime createdAt;
  final List<Defect> defects;
  final List<Rework> reworks;

  Board({
    required this.boardId,
    required this.batch,
    required this.inspectionTime,
    required this.status,
    required this.createdAt,
    this.defects = const [],
    this.reworks = const [],
  });

  factory Board.fromJson(Map<String, dynamic> json) {
    var defectsList = <Defect>[];
    if (json['defects'] != null) {
      defectsList = (json['defects'] as List)
          .map((d) => Defect.fromJson(d as Map<String, dynamic>))
          .toList();
    }
    
    var reworksList = <Rework>[];
    if (json['reworks'] != null) {
      reworksList = (json['reworks'] as List)
          .map((r) => Rework.fromJson(r as Map<String, dynamic>))
          .toList();
    }

    return Board(
      boardId: json['board_id'] as String,
      batch: json['batch'] as String,
      inspectionTime: DateTime.parse(json['inspection_time'] as String),
      status: json['status'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      defects: defectsList,
      reworks: reworksList,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'board_id': boardId,
      'batch': batch,
      'inspection_time': inspectionTime.toIso8601String(),
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'defects': defects.map((d) => d.toJson()).toList(),
      'reworks': reworks.map((r) => r.toJson()).toList(),
    };
  }
}

File: backend/src/main/java/com/tracex/backend/model/Board.java
package com.tracex.backend.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "boards")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = {"defects", "reworks"})
@EqualsAndHashCode(exclude = {"defects", "reworks"})
public class Board {

    @Id
    @Column(name = "board_id", length = 100)
    private String boardId;

    @Column(nullable = false, length = 100)
    private String batch;

    @Column(name = "inspection_time", nullable = false)
    private LocalDateTime inspectionTime;

    @Column(nullable = false, length = 50)
    private String status; // "Passed" or "Failed"

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Builder.Default
    @OneToMany(mappedBy = "board", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<Defect> defects = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "board", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<ReworkTask> reworks = new ArrayList<>();
}
